sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/AbstractRuleExampleTest.java

```diff
@@ -0,0 +1,111 @@
+import java.util.stream.Stream;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.data.MapEntry;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.jupiter.api.function.ThrowingConsumer;
+import org.sonar.api.batch.fs.TextRange;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.batch.sensor.issue.Issue;
+import org.sonar.plugins.common.Check;
+import org.sonar.plugins.common.InputFileContext;
+import org.sonar.plugins.secrets.api.Match;
+import org.sonar.plugins.secrets.api.PatternMatcher;
+import org.sonar.plugins.secrets.api.SpecificationBasedCheck;
+import org.sonar.plugins.secrets.api.SpecificationLoader;
+import org.sonar.plugins.secrets.configuration.model.Rule;
+import org.sonar.plugins.secrets.configuration.model.RuleExample;
+import org.sonar.plugins.secrets.configuration.model.matching.Matching;
+
+import static org.sonar.plugins.common.TestUtils.asString;
+import static org.sonar.plugins.common.TestUtils.inputFile;
+import static org.sonar.plugins.common.TestUtils.sensorContext;
+import static org.sonar.plugins.secrets.api.SpecificationLoader.DEFAULT_SPECIFICATION_LOCATION;
+
+public abstract class AbstractRuleExampleTest {
+    private final Check check;
+    private final URI configurationFile;
+    private final SpecificationLoader specificationLoader;
+
+    protected AbstractRuleExampleTest(Check check, String configurationFile) {
+        specificationLoader = new SpecificationLoader(DEFAULT_SPECIFICATION_LOCATION, Set.of(configurationFile));
+        this.check = check;
+        try {
+            this.configurationFile = SpecificationLoader.class.getClassLoader()
+                    .getResource(DEFAULT_SPECIFICATION_LOCATION + "/" + configurationFile).toURI();
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+        ((SpecificationBasedCheck) check).initialize(specificationLoader);
+    }
+
+    @TestFactory
+    @DisplayName("Execute examples from the configuration file")
+    Stream<DynamicTest> loadExamples() {
+        Stream<MapEntry<Rule, RuleExample>> input = specificationLoader.getAllRules().stream()
+                .flatMap(rule -> rule.getExamples().stream().map(e -> MapEntry.entry(rule, e)));
+        Function<MapEntry<Rule, RuleExample>, String> displayNameGenerator = (pair) ->
+                pair.getKey().getId() + " " + pair.getKey().getMetadata().getName() + " (" + (pair.getValue().isContainsSecret() ? "positive" : "negative") + ")";
+        ThrowingConsumer<MapEntry<Rule, RuleExample>> testExecutor = ruleToExample -> checkExample(ruleToExample.getKey(), ruleToExample.getValue());
+
+        return input
+                .filter(e ->
+                        // TODO: remove this line after tests are fixed
+                        !e.getKey().getId().startsWith("azure-subscription-keys")
```

Comment:
This can be used as a workaround to exclude examples that are not yet working properly (even though it looks like this particular example has some problems with the pattern). Neither Junit nor maven-surefire-plugin support the exclusion of tests by name.

Reply:
Apparently, patterns look good, there must be a flaw somewhere in my logic

Reply:
These are examples for `CommonKeyCheck`, which wasn't active.. All good after moving to grouping by rspecKey instead of configuration file